akka-stream/src/main/scala/akka/stream/scaladsl/SourceWithContext.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2014-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.scaladsl
+
+import scala.annotation.unchecked.uncheckedVariance
+
+import akka.stream._
+import akka.stream.impl.LinearTraversalBuilder
+
+object SourceWithContext {
+  def apply[Out, Mat](underlying: Source[Out, Mat]): SourceWithContext[Out, Out, Mat] = {
+    val under = underlying.map(e ⇒ (e, e))
+    new SourceWithContext[Out, Out, Mat](under, under.traversalBuilder, under.shape)
+  }
+  def from[Out, Ctx, Mat](under: Source[(Out, Ctx), Mat]): SourceWithContext[Ctx, Out, Mat] = {
+    new SourceWithContext[Ctx, Out, Mat](under, under.traversalBuilder, under.shape)
+  }
+}
+
+final class SourceWithContext[+Ctx, +Out, +Mat](
+  underlying:                    Source[(Out, Ctx), Mat],
+  override val traversalBuilder: LinearTraversalBuilder,
+  override val shape:            SourceShape[(Out, Ctx)]
+) extends FlowWithContextOps[Ctx, Out, Mat] with Graph[SourceShape[(Out, Ctx)], Mat] {
+
+  override def withAttributes(attr: Attributes): Repr[Ctx, Out] = new SourceWithContext(underlying, traversalBuilder.setAttributes(attr), shape)
+
+  override type Repr[+C, +O] = SourceWithContext[C, O, Mat @uncheckedVariance]
+  override type Prov[+C, +O] = Source[(O, C), Mat @uncheckedVariance]
+
+  override def via[Ctx2, Out2, Mat2](viaFlow: Graph[FlowShape[(Out, Ctx), (Out2, Ctx2)], Mat2]): Repr[Ctx2, Out2] = from(flow.via(viaFlow))
```

Comment:
`flow.via(viaFlow)` can be replace to `viaFlow` since `flow` here is a unit flow.

Reply:
Thanks!